File: Easy/0027_remove-element/remove-element.py
#!/bin/python3

from typing import List, Tuple

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        for j in range(len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i

def main() -> None :
    tests: List[Tuple[List[int], int]] = [
        ([3,2,2,3],3),
        ([0,1,2,2,3,0,4,2],2),
        ([3],3),
    ]

    for arr, val in tests:
        print("---\narr: ", arr, "\nval: ", val)
        val = Solution().removeElement(arr,val)
        print("--after_solution--", "\nres: ", val)
        print("resArr: ", arr[:val])

if __name__ == "__main__":
    main()

File: Easy/0027_remove-element/remove-element.go
package main

import (
	"fmt"
)

func removeElement(nums []int, val int) int {
	i := 0
	for _, n := range nums {
		if (n != val) {
			nums[i] = n
			i++
		}
	}
	return i
}

type testCase struct {
	arr []int
	val int
}

func main() {
	tests := []testCase{
	      {arr: []int{3,2,2,3}, val: 3},
	      {arr: []int{0,1,2,2,3,0,4,2}, val: 2},
	      {arr: []int{3}, val: 3},
	}

	for _, tc := range tests {
		fmt.Println("---")
		fmt.Printf("arr: %v\nval: %d\n", tc.arr, tc.val)

		res := removeElement(tc.arr, tc.val)
		fmt.Printf("--after_solution--\nres: %d\nresArr: %v\n", res, tc.arr[:res])
	}
}
